File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        unordered_map<int,int> mp;
        for(int val : arr){
            mp[val]++;
        }
        int ans = -1;
        for(auto &it : mp){
            if(it.first == it.second){
                ans = max(ans, it.first);
            }
        }
        return ans;
    }
};

File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        int n = arr.length;

        HashMap<Integer,Integer> mp = new HashMap<>();

        for(int val : arr){
            mp.put(val, mp.getOrDefault(val, 0)+1);
        }
        int ans = -1;
        for(Map.Entry<Integer, Integer> entry : mp.entrySet()){
            if(entry.getValue() == entry.getKey()){
                ans = Math.max(ans, entry.getKey());
            }
        }
        return ans;
    }
}